src/Nethermind/Nethermind.Blockchain/Processing/BlockProcessor.BlockValidationTransactionsExecutor.cs

```diff
@@ -57,7 +60,15 @@ public TxReceipt[] ProcessTransactions(Block block, ProcessingOptions processing
         
             private void ProcessTransaction(Block block, Transaction currentTx, int index, BlockReceiptsTracer receiptsTracer, ProcessingOptions processingOptions)
             {
-                _transactionProcessor.ProcessTransaction(block, currentTx, receiptsTracer, processingOptions, _stateProvider);
+                if ((processingOptions & ProcessingOptions.Trace) != ProcessingOptions.None)
+                {
+                    _callAndRestoreTransactionProcessor.ProcessTransaction(block, currentTx, receiptsTracer, processingOptions, _stateProvider);
+                }
+                else
+                {
+                    _executeTransactionProcessor.ProcessTransaction(block, currentTx, receiptsTracer, processingOptions, _stateProvider);
+                }
+                
```

Comment:
Would prefer either BlockTraceTransactionsExecutor, we can have a base class which can take ITransactionProcessorAdapter